programs/server/config.xml

```diff
@@ -66,69 +66,289 @@
-         See also 'interserver_https_port'.
+
+    <!-- END: Obsolete legacy configurations. Use interface definitions in 'interfaces' instead! -->>
+
+
+
+    <!-- Trusted proxy definitions for specifying in listening interface configs below.
+         See 'allow_proxies' and 'allow_direct' in relevant entries inside 'interfaces' section.
+         Each section can have arbitrary name but names must be unique.
+         The names can be used in 'allow_proxies' section of an interface config.
+         Each section have one mandatory parameter - 'protocol'.
+         Supported protocols are:
+            - PROXY
+            - HTTP
+         You can define any number of proxies for each of the protocols.
       -->
-    <interserver_http_port>9009</interserver_http_port>
+    <proxies>
+
+        <my_load_balancer>
+            <!-- HAProxy PROXY v1/v2 protocol. -->
+            <protocol>PROXY</protocol>
+
+            <!-- The version of the protocol to expect. Can be either 1 or 2.
+                 If not specified, both versions will be allowed and the actual version will be detected automatically.
+              -->
+            <!-- <version>2</version> -->
+
+            <!-- List of IPv4 and/or IPv6 networks, connections from which will be assumed to be from trusted proxies. -->
+            <trust>
+                <net>10.10.10.1/24</net>
+            </trust>
+        </my_load_balancer>
 
-    <!-- Port for communication between replicas with TLS.
-         You have to configure certificate to enable this interface.
-         See the openSSL section below.
-         See also 'interserver_http_credentials'.
+        <my_http_proxy>
+            <!-- Proxy information from X-Forwarded-For HTTP header.-->
+            <protocol>HTTP</protocol>
+
+            <!-- List of IPv4 and/or IPv6 networks, connections from which will be assumed to be from trusted proxies.
+                 Note, that all intermediate peers (but not the initiator client peer) from X-Forwarded-For must be from
+                  some of the networks from this list, for the information about the forwarding chain to be considered authentic.
+              -->
+            <trust>
+                <net>10.10.20.1/24</net>
+            </trust>
+        </my_http_proxy>
+
+    </proxies>
+
+
+    <!-- Interface definitions for ClickHouse server to listen for.
+         Each section can have arbitrary name but names must be unique.
+         The names will be used in logging, as keys for stats, etc.
+         Each section have one mandatory parameter - 'protocol'.
+         Supported protocols are:
+            - native_tcp
+            - native_http
+            - native_grpc
+            - mysql
+            - postgresql
+            - prometheus
+            - interserver_http
+            - keeper_tcp
+         You can define any number of interfaces for each of the protocols.
```

Comment:
So it will be possible to open several http protocols for ex.? 

Several `interserver_http` may be confusing (that port it passed to the zookeeper)

Reply:
Any limitation can be added to the code on per-protocol basis. In general, there is no limit on the number of different listening interfaces for a protocol.

Reply:
I would limit that to a single  interserverver http.